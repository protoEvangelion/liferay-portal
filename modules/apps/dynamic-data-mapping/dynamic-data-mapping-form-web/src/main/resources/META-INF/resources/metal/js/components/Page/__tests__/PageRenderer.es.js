import {dom as MetalTestUtil} from 'metal-dom';

import './__fixtures__/Fields.es';
import Context from './__mock__/mockContext.es';
import PageRenderer from '../PageRenderer.es';

const spritemap = 'icons.svg';
let component;
let componentProps = null;
let context = null;

describe(
	'PageRenderer',
	() => {
		beforeEach(
			() => {
				context = JSON.parse(JSON.stringify(Context));
				componentProps = {
					activePage: 0,
					contentRenderer: 'grid',
					editable: true,
					page: context,
					pageId: 0,
					spritemap,
					total: 1
				};

				jest.useFakeTimers();
			}
		);

		it(
			'should resize a specific column on the drag and drop layout',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				component._handleOnClickResize();
				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should display empty drag message when there are rows with no columns specified',
			() => {
				component = new PageRenderer(
					{
						...componentProps,
						page: {
							rows: [{}]
						}
					}
				);
				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should change the page title',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				const pageTitle = component.element.querySelector('.form-builder-page-header-title');
				const spy = jest.spyOn(component, 'emit');

				pageTitle.value = 'Page Title';

				jest.runAllTimers();
				MetalTestUtil.triggerEvent(pageTitle, 'keyup', {});

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('updatePage', expect.any(Object));

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should change the page title',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				const pageDescription = component.element.querySelector('.form-builder-page-header-description');
				const spy = jest.spyOn(component, 'emit');

				pageDescription.value = 'Page Description';

				jest.runAllTimers();
				MetalTestUtil.triggerEvent(pageDescription, 'keyup', {});

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('updatePage', expect.any(Object));

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render a layout and emit an event when delete button is clicked',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				const spy = jest.spyOn(component, 'emit');

				component.element.querySelector('button[aria-label=\'trash\']').click();
				component.element.querySelector('.modal .btn-primary').click();

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('deleteButtonClicked', expect.any(Object));
			}
		);

		it(
			'should render a layout and emit an event when duplicate button is clicked',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				const spy = jest.spyOn(component, 'emit');

				component.element.querySelector('button[aria-label=\'paste\']').click();

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('duplicateButtonClicked', expect.any(Object));
			}
		);

		it(
			'should render a layout and continue to propagate the field edit event',
			() => {
				component = new PageRenderer(
					{
						...componentProps,
						contentRenderer: 'list'
					}
				);

				const spy = jest.spyOn(component, 'emit');

				jest.runAllTimers();
				component._handleFieldChanged();

				expect(spy).toHaveBeenCalled();
			}
		);

		it(
			'should render a layout with emit an field clicked event',
			() => {
				component = new PageRenderer(
					{
						...componentProps
					}
				);

				const spy = jest.spyOn(component, 'emit');

				component.element.querySelector('.ddm-drag').click();

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith('fieldClicked', expect.any(Object));
			}
		);

		it(
			'should emit a fieldClicked event with the field location',
			() => {
				component = new PageRenderer(
					{
						...componentProps,
						dragAndDropDisabled: true
					}
				);

				const spy = jest.spyOn(component, 'emit');

				component.element.querySelector('.ddm-drag').click();

				expect(spy).toHaveBeenCalled();
				expect(spy).toHaveBeenCalledWith(
					'fieldClicked',
					expect.objectContaining(
						{
							columnIndex: expect.anything(),
							pageIndex: expect.any(Number),
							rowIndex: expect.any(Number)
						}
					)
				);
			}
		);
	}
);