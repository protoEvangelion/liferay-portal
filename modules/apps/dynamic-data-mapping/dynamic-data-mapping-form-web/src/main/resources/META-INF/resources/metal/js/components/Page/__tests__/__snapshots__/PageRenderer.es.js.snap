// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRenderer should change the page title 1`] = `
<div class="ddm-page-container-layout">
  <div class="ddm-form-page pl-3 pr-3">
    <textarea class="form-builder-page-header-title form-control" data-page-id="0" maxlength="120" value="Untitled name" placeholder="untitled-page-x-of-x"></textarea>
    <textarea class="form-builder-page-header-description form-control" data-page-id="0" maxlength="120" placeholder="add-a-short-description-for-this-page"></textarea>
    <div class="form-builder-layout">
      <div class="layout-page">
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="0" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-3 col-ddm" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-9" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="1" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-4 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-6 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-2" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="2">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="2" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12 col-ddm" data-ddm-field-row="2" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="3" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="fade modal" role="dialog" tabindex="-1">
    <div class="modal-dialog modal-sm">
      <div class="modal-content">
        <div class="modal-header">
          <div class="modal-title">delete-field-dialog-title</div>
          <button class="btn close btn-unstyled" aria-label="close" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times" focusable="false">
              <title>times</title>
              <use xlink:href="icons.svg#times"></use>
            </svg>
          </button>
        </div>
        <div class="modal-body">
          <h4>are-you-sure-you-want-to-delete-this-field</h4>
        </div>
        <div class="modal-footer">
          <div class="modal-item-first">
            <div class="btn-group"></div>
          </div>
          <div class="modal-item-last">
            <div class="btn-group">
              <div class="btn-group-item">
                <button class="btn close-modal btn-secondary" type="button">dismiss</button>
              </div>
              <div class="btn-group-item">
                <button class="btn btn-primary" type="button">delete</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageRenderer should change the page title 2`] = `
<div class="ddm-page-container-layout">
  <div class="ddm-form-page pl-3 pr-3">
    <textarea class="form-builder-page-header-title form-control" data-page-id="0" maxlength="120" value="Untitled name" placeholder="untitled-page-x-of-x"></textarea>
    <textarea class="form-builder-page-header-description form-control" data-page-id="0" maxlength="120" placeholder="add-a-short-description-for-this-page"></textarea>
    <div class="form-builder-layout">
      <div class="layout-page">
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="0" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-3 col-ddm" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-9" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="1" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-4 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-6 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-2" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="2">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="2" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12 col-ddm" data-ddm-field-row="2" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="3" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="fade modal" role="dialog" tabindex="-1">
    <div class="modal-dialog modal-sm">
      <div class="modal-content">
        <div class="modal-header">
          <div class="modal-title">delete-field-dialog-title</div>
          <button class="btn close btn-unstyled" aria-label="close" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times" focusable="false">
              <title>times</title>
              <use xlink:href="icons.svg#times"></use>
            </svg>
          </button>
        </div>
        <div class="modal-body">
          <h4>are-you-sure-you-want-to-delete-this-field</h4>
        </div>
        <div class="modal-footer">
          <div class="modal-item-first">
            <div class="btn-group"></div>
          </div>
          <div class="modal-item-last">
            <div class="btn-group">
              <div class="btn-group-item">
                <button class="btn close-modal btn-secondary" type="button">dismiss</button>
              </div>
              <div class="btn-group-item">
                <button class="btn btn-primary" type="button">delete</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageRenderer should display empty drag message when there are rows with no columns specified 1`] = `
<div class="ddm-page-container-layout">
  <div class="ddm-form-page pl-3 pr-3">
    <textarea class="form-builder-page-header-title form-control" data-page-id="0" maxlength="120"></textarea>
    <textarea class="form-builder-page-header-description form-control" data-page-id="0" maxlength="120" placeholder="add-a-short-description-for-this-page"></textarea>
    <div class="form-builder-layout">
      <div class="layout-page">
        <div class="col col-empty col-md-12 last-col lfr-initial-col">
          <div id="lfr-initial-col-message">drag-from-sidebar-and-drop-here</div>
        </div>
      </div>
    </div>
  </div>
  <div class="fade modal" role="dialog" tabindex="-1">
    <div class="modal-dialog modal-sm">
      <div class="modal-content">
        <div class="modal-header">
          <div class="modal-title">delete-field-dialog-title</div>
          <button class="btn close btn-unstyled" aria-label="close" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times" focusable="false">
              <title>times</title>
              <use xlink:href="icons.svg#times"></use>
            </svg>
          </button>
        </div>
        <div class="modal-body">
          <h4>are-you-sure-you-want-to-delete-this-field</h4>
        </div>
        <div class="modal-footer">
          <div class="modal-item-first">
            <div class="btn-group"></div>
          </div>
          <div class="modal-item-last">
            <div class="btn-group">
              <div class="btn-group-item">
                <button class="btn close-modal btn-secondary" type="button">dismiss</button>
              </div>
              <div class="btn-group-item">
                <button class="btn btn-primary" type="button">delete</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PageRenderer should resize a specific column on the drag and drop layout 1`] = `
<div class="ddm-page-container-layout">
  <div class="ddm-form-page pl-3 pr-3">
    <textarea class="form-builder-page-header-title form-control" data-page-id="0" maxlength="120" value="Untitled name"></textarea>
    <textarea class="form-builder-page-header-description form-control" data-page-id="0" maxlength="120" placeholder="add-a-short-description-for-this-page"></textarea>
    <div class="form-builder-layout">
      <div class="layout-page">
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="0" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-3 col-ddm" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-9" data-ddm-field-row="0" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="1" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-4 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-6 col-ddm" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="1">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
          <div class="col-md-2" data-ddm-field-row="1" data-ddm-field-page="0" data-ddm-field-column="2">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="2" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12 col-ddm" data-ddm-field-row="2" data-ddm-field-page="0" data-ddm-field-column="0">
            <div class="ddm-field-container">
              <div class="ddm-resize-handle-left"></div>
              <div class="ddm-drag">
                <div class="ddm-field-not-found">Field doesn't exist.</div>
              </div>
              <div class="ddm-field-actions-container">
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="paste" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-paste" focusable="false">
                    <title>paste</title>
                    <use xlink:href="icons.svg#paste"></use>
                  </svg>
                </button>
                <button class="btn btn-monospaced btn-sm btn-secondary" aria-label="trash" type="button">
                  <svg aria-hidden="true" class="lexicon-icon lexicon-icon-trash" focusable="false">
                    <title>trash</title>
                    <use xlink:href="icons.svg#trash"></use>
                  </svg>
                </button>
              </div>
              <div class="ddm-resize-handle-right"></div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col-md-12" data-ddm-field-row="3" data-ddm-field-page="0">
            <div class="col-empty ddm-target"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="fade modal" role="dialog" tabindex="-1">
    <div class="modal-dialog modal-sm">
      <div class="modal-content">
        <div class="modal-header">
          <div class="modal-title">delete-field-dialog-title</div>
          <button class="btn close btn-unstyled" aria-label="close" type="button">
            <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times" focusable="false">
              <title>times</title>
              <use xlink:href="icons.svg#times"></use>
            </svg>
          </button>
        </div>
        <div class="modal-body">
          <h4>are-you-sure-you-want-to-delete-this-field</h4>
        </div>
        <div class="modal-footer">
          <div class="modal-item-first">
            <div class="btn-group"></div>
          </div>
          <div class="modal-item-last">
            <div class="btn-group">
              <div class="btn-group-item">
                <button class="btn close-modal btn-secondary" type="button">dismiss</button>
              </div>
              <div class="btn-group-item">
                <button class="btn btn-primary" type="button">delete</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
